### PR TITLE
[trusted state] replace version with waypoint

### PR DESCRIPTION
**Pull request by zekun000**
## Motivation

This PR fix the problem that we don't verify latest ledger info with the
same version as local.

Now the latest ledger info has to be consistent with local verified
state or carry valid signatures to be trusted.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

Existing tests.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

File: types/src/unit_tests/trusted_state_test.rs [hunk 1]
```diff
@@ -402,21 +402,29 @@ proptest! {
...
+        // signatures and invalid data should fail.
+        let bad_li = LedgerInfoWithSignatures::new(
+            LedgerInfo::new(
+                BlockInfo::new(
+                    good_li.epoch(),
+                    0,                 /* round */
+                    HashValue::zero(), /* id */
+                    HashValue::zero(), /* executed_state_id */
+                    good_li.version(),
+                    42, /* bad timestamp_usecs */
```
Comment (msmouse) [inline types/src/unit_tests/trusted_state_test.rs:417]: This changes the purpose of the test and renders the name of the test confusing..  I'd rather keep it the old way. Probably add new test just for the "latest li at the same version with my trusted state" case.

Comment (zekun000) [reply types/src/unit_tests/trusted_state_test.rs:417]: oh forget to change the name, how about failes_with_invalid_latest_li

**Comment by msmouse** [comment]
hmm.. let `Waypoint` wrap a `Checkpoint` will please me, not sure about others :D

File: types/src/unit_tests/trusted_state_test.rs [hunk 2]
```diff
@@ -402,21 +402,29 @@ proptest! {
...
         let initial_li = initial_li_with_sigs.ledger_info();
         let trusted_state = TrustedState::try_from(initial_li).unwrap();
+        let good_li = latest_li.ledger_info();
 
         // Verifying a latest ledger info (inside the last epoch) with invalid
-        // signatures should fail.
-        let latest_li = LedgerInfoWithSignatures::new(
-            latest_li.ledger_info().clone(),
-            BTreeMap::new(), /* empty signatures */
+        // signatures and invalid data should fail.
```
Comment (davidiw) [inline types/src/unit_tests/trusted_state_test.rs:408]: can we actually test both?

**Comment by zekun000** [comment]
@davidiw for waypoint access, it's https://developers.libra.org/testnet_waypoint.txt and release oncall will create PR and let eric merge it

**Comment by davidiw** [comment]
How do we generate a waypoint? Is this going to use config-builder for the near future? How would I generate one without config-builder?

**Comment by davidiw** [comment]
Let's consider adding to the config/README.md how to generate a waypoint

@bors-libra delegate+

**Pull request closed without merging**